testdata/testdata.go

```diff
@@ -613,3 +622,15 @@ func notUsed_Also_Used_In_Else_Body_OK() {
 		noOp1(x)
 	}
 }
+
+func varUsedInTwoIfStmts_OK() {
+	hasFlag := getBool()
+	if hasFlag {
+		noOp1()
+	}
+	a := 0
+	if hasFlag {
+		noOp2()
+	}
+	noOp1(a)
```

Comment:
Strangely enough, putting the two if statements back to back make it work. 
``` go
if hasFlag {}
if hasFlag {}
```
You need an expression in between to make it fail. Also, altering the second if to `hasFlag && a == 0` makes it work again. I was wondering if maybe the analyzer got some optimized version of the AST, but when I used my debugger I found that it gives the AST as is.